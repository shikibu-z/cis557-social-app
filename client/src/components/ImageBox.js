/* eslint-disable no-unneeded-ternary */
/* eslint-disable no-nested-ternary */
import { React, useState } from 'react';
import '../stylesheets/ImageBox.css';

function ImageBox(props) {
  const { image } = props;
  const [file, setFile] = useState(null);

  const fileHandler = (e) => {
    setFile(e.target.files[0]);
  };

  function handleClick() {
    document.getElementById('imgInp').click();
  }

  return (
    <div>
      <form id="form1" runat="server">
        <input onChange={fileHandler} type="file" id="imgInp" />
        <div className="image-container">
          <img onClick={handleClick} role="none" onKeyPress={handleClick} id="image" src={file ? URL.createObjectURL(file) : (image ? image : null)} alt="" height="300px" width="300px" style={{ 'object-fit': 'cover' }} />
        </div>
      </form>
    </div>
  );
}

export default ImageBox;
