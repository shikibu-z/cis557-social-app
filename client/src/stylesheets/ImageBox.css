.image-container{
    position: relative;
    width: 50%;
    max-width: 150px;
}

#imgInp {
    display: none;
}

#image {
    border-radius: 50%;
    width: 150px;
    height: 150px;
    background-color: white;
    border: 1px solid #F8B114;
    transition: .5s ease;
}

.overlay {
    transition: .5s ease;
    opacity: 0;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    -ms-transform: translate(-50%, -50%);
    text-align: center;
}
.image-container:hover #image{
    opacity: 0.3;
}

.image-container:hover .overlay{
    opacity: 1;
}