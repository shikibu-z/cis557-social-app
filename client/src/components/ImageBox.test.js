/*eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import ImageBox from './ImageBox';

test('Image box matches snapshot', () => {
  const component = renderer.create(<ImageBox />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
