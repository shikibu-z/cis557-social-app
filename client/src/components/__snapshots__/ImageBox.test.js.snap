// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image box matches snapshot 1`] = `
<div>
  <form
    id="form1"
    runat="server"
  >
    <input
      id="imgInp"
      onChange={[Function]}
      type="file"
    />
    <div
      className="image-container"
    >
      <img
        alt=""
        height="300px"
        id="image"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="none"
        src={null}
        style={
          Object {
            "object-fit": "cover",
          }
        }
        width="300px"
      />
    </div>
  </form>
</div>
`;
